moveit_core/robot_trajectory/include/moveit/robot_trajectory/robot_trajectory.h

```diff
@@ -144,32 +145,57 @@ class RobotTrajectory
     return waypoints_.front();
   }
 
-  const std::deque<double>& getWayPointDurations() const
+  std::deque<rclcpp::Duration> const& getDurationFromPreviousDeque() const
   {
     return duration_from_previous_;
   }
 
+  [[deprecated("getWayPointDurations is deprecated, replaed by getDurationFromPreviousDeque")]] std::deque<double>
+  getWayPointDurations() const
+  {
+    std::deque<double> ret;
+    std::transform(duration_from_previous_.cbegin(), duration_from_previous_.cend(), ret.begin(),
+                   [](auto const& e) { return e.seconds(); });
+    return ret;
+  }
+
   /** @brief  Returns the duration after start that a waypoint will be reached.
    *  @param  The waypoint index.
    *  @return The duration from start; returns overall duration if index is out of range.
    */
-  double getWayPointDurationFromStart(std::size_t index) const;
+  rclcpp::Duration getWayPointDurationFromStartAt(std::size_t index) const;
```

Comment:
Long term I'd like to make all these free functions. 

Maybe I should just design that and we can discuss that plan instead of this incremental PR.

The reason I made this PR was to separate the idea of going from double -> rclcpp::Duration from a larger data-structure change.